### PR TITLE
Marco/feature/onboarding

### PR DESCRIPTION
**Pull request by MarcoFrancisco13**
- [x] Login Page
- [x] Onboarding Screens

image

image

image

### CODE REVIEW

File: lib/screens/login.dart [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/material.dart';
+import 'package:admu_student_app/screens/onboarding.dart';
+
+class login extends StatelessWidget {
```
Comment (devnamme) [inline lib/screens/login.dart]: class names usually start with capital and use camelcase: `class Login extends...` or `class BottomNavBar extends ...`

File: lib/screens/onboarding.dart [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          Image(image: AssetImage('assets/onboarding.png'), fit: BoxFit.fill),
+    ),
+    decoration: const PageDecoration(
+        titleTextStyle: TextStyle(fontWeight: FontWeight.bold, fontSize: 40),
+        bodyTextStyle: TextStyle(fontSize: 15),
+        imageFlex: 2),
+  )
+];
+
+class _onboardingState extends State<onboarding> {
```
Comment (devnamme) [inline lib/screens/onboarding.dart]: also here: `class _OnboardingState extends State<Onboarding>`
change the declaration for `class Onboarding` above

**Comment by devnamme** [comment]
try using the font from the [google_fonts package](https://pub.dev/packages/google_fonts), as far as i know, `DM Sans` is the font that we're using

**Comment by devnamme** [comment]
if possible, also try to fix the spacing for the login page and perhaps convert the onboarding images to plain shapes (it has a padding in the left and right edges since it only uses an image

tho for the latter, i don't know how that package/widget works. in case that it's not possible to remove the padding, you can leave it for now since the onboarding page isn't finalized yet from the design

**Comment by MarcoFrancisco13** [comment]
- [x] Fixed Padding in Login Page
- [x] Changed onboarding screens to solid colors
- [x] Added DM Sans font in all screens

image

image

File: lib/screens/login.dart [hunk 3]
```diff
@@ -0,0 +1,156 @@
+import 'package:flutter/material.dart';
+import 'package:admu_student_app/screens/onboarding.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+class Login extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MediaQuery(
+      data: MediaQueryData(),
+      child: MaterialApp(
```
Comment (devnamme) [inline lib/screens/login.dart]: `lib/main.dart` already uses a `MaterialApp` class to generate the app, having another one (nested) might lead to errors later on (?)

File: lib/screens/login.dart [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+              child: Container(
+                //Parent Column
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    CircleAvatar(
+                      backgroundImage: AssetImage('onboarding.png'),
+                      radius: 100,
+                    ),
+                    Text('Welcome to Beadle!',
```
Comment (devnamme) [inline lib/screens/login.dart]: text isn't centered (visible in smaller screens)
image

File: lib/screens/onboarding.dart [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import 'package:flutter/material.dart';
```
Comment (devnamme) [inline lib/screens/onboarding.dart:1]: although this isn't really a standard, should we stick to a simple format for imports?

```text
dart

flutter and other packages

project
```

example: 
```dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:google_fonts/...';

import 'package:admu_student_app/...';
```

**Review by devnamme — APPROVED** [review]
small changes only, the issue with `MaterialApp` can be changed later on when we receive our app icon. you can merge it (after fixing conflicts), let me know if it's done to fix some stuff in develop

File: lib/screens/onboarding.dart (hunk 5 above)
Comment (MarcoFrancisco13) [reply lib/screens/onboarding.dart:1]: got it will make changes in a bit !

**Comment by MarcoFrancisco13** [comment]
- [x] Fixed centering issue on smaller screens
- [x] Arranged imports according to format 

image

**Review by devnamme — APPROVED** [review]
great job dude! after merging this, the develop branch will still be broken (conflicts, class names, navigation, etc). i'll also let Chris know later in the gc so that we can hopefully have a working dev branch by Tuesday night - which means that creating a new branch before that one will lead to more conflicts. i'll also be sending some changes for the tasks/timeline this week (moving deadlines, etc.)

love those colors in the onboarding screens btw!

**Comment by devnamme** [comment]
hello! should i merge this now (and try to fix the conflicts) or will you be the one doing it?
this is the first time seeing that "too complex to resolve in the web editor" message

**Comment by MarcoFrancisco13** [comment]
> 
> 
> hello! should i merge this now (and try to fix the conflicts) or will you be the one doing it?
> this is the first time seeing that "too complex to resolve in the web editor" message

Sorry late reply! I'm not too sure how to resolve the conflicts also, but I'll see if I can finish it all before our meeting tomorrow

**Comment by devnamme** [comment]
hmm, i'm guessing that the problem is with `lib/main.dart` since dev has it in the `lib` folder while yours is in `lib/screens` (but that was already fixed before). but thanks! i might also run some tests on my local branch to find out how to fix it

**Pull request merged**